Models/commentModel.js

```diff
@@ -1,31 +1,18 @@
 const db = require("../db/connection");
 
 exports.selectCommentsByArticleId = (article_id) => {
-    const idNum = parseInt(article_id)
-    if (!Number.isNaN(idNum)) {
-        return db
-            .query(`SELECT *
-                    FROM comments
-                    WHERE article_id = $1;`, [article_id])
-            .then((result) => {
-                if (result.rows.length === 0) {
-                    return Promise.reject({
-                        status: 404,
-                        msg: `No comments found for article_id: ${article_id}`
-                    });
-                } else {
-                    return db.query(`SELECT *
-                                     FROM comments
-                                     WHERE article_id = $1;`, [article_id])
-                        .then((response) => {
-                            return response.rows
-                        })
-                }
-            })
-    } else {
-        return Promise.reject({
-            status: 400,
-            msg: 'bad request'
-        })
+    const queryValues = [article_id];
+    let queryStr = "SELECT * FROM comments WHERE article_id = $1";
+    if (isNaN(article_id)) {
+        return Promise.reject({status: 400, msg: "bad request"});
     }
-}
+    return db.query(queryStr, queryValues).then((result) => {
+        if (!result.rows.length) {
+            return Promise.reject({
+                status: 404,
+                msg: `No comments found for article_id: ${article_id}`,
+            });
+        }
```

Comment:
Ideally we would want to send back an empty array if there is no comments but the article DOES exist, and a 404 if the article doesnt exist and has no comments. Im happy for you to move on from this but just remember you will have to deal with this in your front end when displaying articles with 0 comments

Reply:
i didnt understand how to do that so i put in an nc help at 12 10 but got no response